OpenRA.Mods.D2k/Traits/World/VariedCostManager.cs

```diff
@@ -0,0 +1,53 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Traits
```

Comment:
This is Generic enough imo. I'm not sure if it should live in the D2k dll.

Reply:
Moved to Common then.